app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,63 @@
+class TasksController < ApplicationController
+
+  def index
+    @tasks = Task.all
+  end
+
+  def show
+    @task = Task.find(params[:id])
+    @datetime = @task.completed_at
+
+    if @task.completed_at != nil
+      @status = "Done at #{@datetime.strftime('%I:%M %P')} on #{@datetime.strftime('%B %e, %Y')}"
+    else
+      @status = "Still in progress"
+    end
+  end
+
+  def new
+    @task = Task.new
+  end
+
+  def create
+    @task = Task.new
+
+    @task.title = params[:task][:title]
+    @task.description = params[:task][:description]
+    @task.completed_at = params[:task][:completed_at]
+
+    @task.save
+
+    redirect_to root_path
+  end
+
+  def edit
+    @task = Task.find(params[:id])
+  end
+
+  def button
```

Comment:
@brianaeng You might want to find a better name for the action that marks a task as complete rather than "button."  I'm just saying...